woocommerce/woocommerce-gutenberg-products-block
# Refactor product-control to TypeScript, clean up related types.

## What

Convert product-control to TypeScript, remove prop types and clean up related types.

Fixes #9537

## Why

This is desired to refactor existing JS components to TS.

I've opted not to add additional tests here, there is test coverage of both the HOCs it uses and other underlying components and we can also perform some manual testing of the block.

## Testing Instructions

1. A smoke test can involve creating a page and testing the functionality of the single product block, e.g. search for product, choose product and check that it still works. 


## WooCommerce Visibility


Required:


## Checklist

Required:

Conditional:

## Changelog


> Add suggested changelog entry here.
